tasks/update_appraisal_gemfiles.rake

```diff
@@ -42,7 +55,15 @@ task :update_appraisal_gemfiles do |_task, args|
   versions = tracer_version_arg.empty? ? TRACER_VERSIONS : tracer_version_arg
 
   versions.each do |version|
+    forced_bundler_version = nil
+    bundler_setup = 'gem update bundler'
+
+    if FORCE_BUNDLER_VERSION.include?(version)
+      forced_bundler_version = "_#{FORCE_BUNDLER_VERSION[version]}_"
+      bundler_setup = "gem install bundler -v #{FORCE_BUNDLER_VERSION[version]}"
+    end
+
     sh "docker-compose run -e APPRAISAL_GROUP --no-deps --rm tracer-#{version} /bin/bash -c " \
-      "'rm -f Gemfile.lock && gem update bundler && bundle install && bundle exec appraisal update'"
+      "'rm -f Gemfile.lock && #{bundler_setup} && bundle #{forced_bundler_version} install && bundle exec appraisal #{forced_bundler_version} update'"
```

Comment:
Shouldn't that run `bundle exec appraisal generate` as well?

IIRC gemfiles are not regenerated when the files are found to exist.

Reply:
Additionally, we should ideally run `bundle lock --lockfile <lockfile> --add-platform [x86_64-linux,aarch64-linux]` when possible (I think it's not supported on 1.17.x)

Reply:
@lloeki I think you're right. Would you be up for opening a PR adding that to both tasks?

Reply:
> Additionally, we should ideally run bundle lock --lockfile <lockfile> --add-platform [x86_64-linux,aarch64-linux] when possible (I think it's not supported on 1.17.x)

Actually, @lloeki isn't that blocked on the sorbet changes on the Gemfile?

Reply:
> IIRC gemfiles are not regenerated when the files are found to exist.

I've just tried this and I think they are. As an example, I added an extra gem to one of the existing appraisal gemfiles, then ran `bundle exec rake install_appraisal_gemfiles` and the lockfiles were not changed BUT the gemfile was regenerated and the extra gem was removed.

Reply:
Yes it is, we need to move forward with @marcotc's PR https://github.com/DataDog/dd-trace-rb/pull/2421